src/seedu/addressbook/data/person/Address.java

```diff
@@ -3,57 +3,164 @@
+            + "where block street and unit are nonempty strings while postal\n"
+            + "code is a 6 digit number. They must be separated by commas";
     public static final String ADDRESS_VALIDATION_REGEX = ".+";
+    private static final String ADDRESS_POSTALCODE_VALIDATION_REGEX = "\\d{6}";
+    private static final String ADDRESS_UNIT_VALIDATION_REGEX = ".+";
+    private static final String ADDRESS_STREET_VALIDATION_REGEX = ".+";
+    private static final String ADDRESS_BLOCK_VALIDATION_REGEX = ".+";
+
+    private AddressBlock addressBlock;
+    private AddressStreet addressStreet;
+    private AddressUnit addressUnit;
+    private AddressPostalcode addressPostalcode;
 
-    public final String value;
     private boolean isPrivate;
 
     /**
      * Validates given address.
      *
-     * @throws IllegalValueException if given address string is invalid.
+     * @throws IllegalValueException
+     *             if given address string is invalid.
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
         String trimmedAddress = address.trim();
         this.isPrivate = isPrivate;
         if (!isValidAddress(trimmedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = trimmedAddress;
+        initialisePersonData(trimmedAddress);
+    }
+
+
+    /**
+     * Initialises the attributes of the Address object by splitting the string using comma.
+     */
+    private void initialisePersonData(String trimmedAddress) throws IllegalValueException {
+        String[] personDataSplit = trimmedAddress.split(",");
+        if (personDataSplit.length != 4) {
+            throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
+        }
+        this.addressBlock = new AddressBlock(personDataSplit[0]);
+        this.addressStreet = new AddressStreet(personDataSplit[1]);
+        this.addressUnit = new AddressUnit(personDataSplit[2]);
+        this.addressPostalcode = new AddressPostalcode(personDataSplit[3]);
+
+    }
+
+    /**
+     * Returns true if a given string is a valid person's address, which
+     * contains the block, street, unit and postalcode information.
+     */
+    public static boolean isValidAddress(String test) throws IllegalValueException {
+        String[] personDataSplit = test.split(",");
+        if (personDataSplit.length != LENGTH_OF_CORRECT_ADDRESS) {
+            throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
+        }
+        boolean isValidBlock = isValidBlock(personDataSplit[0]);
+        boolean isValidStreet = isValidStreet(personDataSplit[1]);
+        boolean isValidUnit = isValidUnit(personDataSplit[2]);
+        boolean isValidPostalcode = isValidPostalcode(personDataSplit[3]);
+        return isValidBlock && isValidStreet && isValidUnit && isValidPostalcode;
```

Comment:
Can more refactoring be done here?
The boolean variables does not seem to do much except for returning from the method.
Could the boolean variables be then removed?
Wouldn't `return isValidBlock(...) && isValidStreet(...) && isValidUnit(...) && isValidPostalCode(...)` be better?